Lib/test/test_posixpath.py

```diff
@@ -660,6 +660,21 @@ def test_realpath_resolve_first(self):
             safe_rmdir(ABSTFN + "/k")
             safe_rmdir(ABSTFN)
 
+    @skip_if_ABSTFN_contains_backslash
+    def test_realpath_strict_nondir(self):
+        try:
+            with open(ABSTFN, 'w') as f:
+                f.write('test_posixpath wuz ere')
+            self.assertEqual(realpath(ABSTFN, strict=True), ABSTFN)
+            with self.assertRaises(NotADirectoryError):
+                realpath(ABSTFN + "/", strict=True)
+            with self.assertRaises(NotADirectoryError):
+                realpath(ABSTFN + "/.", strict=True)
```

Comment:
And what about `/..`?

Please add tests with the default (strict=False) mode for every tested path here, for comparison.

What if the path is a symlink followed by `/`? `symlink/`.

What if the path is a symlink which is resolved to a path to a non-dir (a file or a symlink followed by `/`)? `symlink` -> `file/` or `symlink` -> `symlink2/`?